web_client/analysis-modules/feature-matching.jsx

```diff
@@ -117,7 +117,115 @@ const computeMatch = (data, page) => {
   );
 };
 
-const applyMatch = (data, page) => { /* TODO(opadron): */ };
+const applyMatch = (data, page) => {
+  const truncatePromise = Promise.all([
+    D(actions.truncateAnalysisPages(1, {
+      clear: true,
+      disable: true,
+      remove: false
+    })),
+
+    (
+      applyResultElements
+        ? D(actions.removeAnalysisElement(applyResultElements))
```

Comment:
This is another case of "non-standard" syntax making readability more difficult. In this case, you have a parenthesized expression set off from its immediate context by a blank line. It therefore takes a mental "zoom-out" to realize that it's in fact the second element in a list of promises, and a unique one at that: sometimes there is no promise to execute, which is represented by a null value that passes through to the eventual callback. I think the idea would be better expressed by something like this:

```javascript
  let promises = [
    D(actions.truncateAnalysisPages(1, {
      clear: true,
      disable: true,
      remove: false
    })),
    null
  ];

  if (applyResultElements) {
    promises[1] = D(actions.removeAnalysisElement(applyResultElements);
  } 

  const truncatePromise = Promise.all(promises);
```

This is not the most elegant JavaScript I've ever written, and it's definitely not wrong in an objective sense to use `?:` here, but I still think my version is more readable. It is ultimately a matter of style, so go with what makes the most sense to you.

The only part of my critique I want to absolutely leave in place is that non-standard spacing/linebreaks/syntax will definitely slow down future readers of this code, and should be avoided.